pkg/tuned/controller.go

```diff
@@ -1029,6 +1029,7 @@ func (c *Controller) changeSyncerTuneD(change Change) (synced bool, err error) {
 			// Cache the value written to tunedRecommendFile.
 			c.daemon.recommendedProfile = change.recommendedProfile
 			klog.V(1).Infof("recommended TuneD profile updated from %q to %q [inplaceUpdate=%v nodeRestart=%v]", prevRecommended, change.recommendedProfile, inplaceUpdate, change.nodeRestart)
+			changeRecommend = true
```

Comment:
Good catch with the variable not being set anywhere in this function.  However, can we possibly get rid of the variable completely?  Would it work if we replaced it further down just by `change.nodeRestart` ?  I haven't tested this (yet), but looking at the code it should work.  It would also resolve the slight mis-placement of the variable setting below the logging message and its not very intuitive name.

Reply:
I was thinking about that, but there are two separate triggers for this flow:

- !inplaceUpdate -- the recommended profile was changed for a live system
- change.nodeRestart -- first execution after reboot

The !inplaceUpdate is actually the case that was missed and caused the bug.

Both variables are later used for flow branching and the changeRecommend bool is the combination of both. I agree it is not easily visible or understandable, but I was a bit afraid to touch the logic tbh.

Reply:
Yeah you're right.  I was just testing that and my suggestion didn't work.  I agree `!inplaceUpdate` causes that issue.  

> I was a bit afraid to touch the logic tbh.

Understood.  Well, I wonder if the entire flow needs a bit of cleanup / adding comments once @ffromani is back.  In the meantime, I guess we can merge if this fixes the issue.